src/traveltime_drive_time_comparisons/outlier_detection.py

```diff
@@ -0,0 +1,201 @@
+"""
+Outlier detection for travel time comparisons.
+
+This module provides outlier detection to identify and filter out rows where one provider
+returns a drastically different travel time compared to others, indicating potential
+data quality issues (e.g., routing errors, API failures).
+"""
+
+import logging
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+import numpy as np
+import pandas as pd
+from pandas import DataFrame
+
+from traveltime_drive_time_comparisons.common import get_capitalized_provider_name
+
+
+@dataclass
+class OutlierConfig:
+    """Configuration for outlier detection."""
+
+    enabled: bool = True
+    """
+    Whether outlier detection is enabled.
+    If False, no outliers will be detected and all rows will be kept.
+    """
+
+    ratio_threshold: float = 3.0
+    """
+    Threshold for flagging outliers as a ratio of difference from median.
+    Values are flagged if their ratio is greater than or equal to this threshold.
+
+    Examples with ratio_threshold=3.0:
+    - Value of 3000 when median is 1000 → ratio = 3.0 → flagged as outlier (exactly at threshold)
+    - Value of 5000 when median is 1000 → ratio = 5.0 → flagged as outlier
+    - Value of 1500 when median is 1000 → ratio = 1.5 → NOT flagged
+    - Value of 300 when median is 1000 → ratio = 3.33 → flagged as outlier
+    - Value of 2999 when median is 1000 → ratio = 2.999 → NOT flagged (just below threshold)
+
+    Recommended values:
+    - 3.0: Catches egregious errors (default, recommended)
+    - 2.0: More strict, catches moderate differences
+    - 5.0: Very lenient, only extreme cases
+
+    Note: With only 2 providers, a value needs to be ~5x different from the other
+    to be flagged with threshold=3.0 (e.g., [5000, 1000] will flag the 1000).
+    """
+
+    min_providers: int = 2
```

Comment:
I think that 3 is a more reasonable default. With 2, what we effectively do is dropping all "difficult" routes. I think it goes beyond the common understanding of what an "outlier" is.

Maybe we should make this configurable with CLI?